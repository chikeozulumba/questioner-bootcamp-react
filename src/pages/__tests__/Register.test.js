import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import RegisterPage from '../register';
import randomEmail from 'random-email';

describe('<RegisterPage />', () => {
  it('should render component when validation fails', () => {
    const registerPage = mount(<BrowserRouter><RegisterPage /></BrowserRouter>);
    expect(registerPage.find('.reg-btn.box-sh-dark.hover-lg')).toHaveLength(1);
    expect(registerPage).toMatchSnapshot();
    registerPage.find('input[name="fullName"]').simulate('change', { target: {name: "fullName", value: "" }});
    registerPage.find('input[name="lastname"]').simulate('change', { target: {name: "lastname", value: "" }});
    registerPage.find('input[name="email"]').simulate('change', { target: {name: "email", value: "" }});
    registerPage.find('input[name="phone"]').simulate('change', { target: {name: "phone", value: "" }});
    registerPage.find('input[name="password"]').simulate('change', { target: {name: "password", value: "" }});
    registerPage.find('.reg-btn.box-sh-dark.hover-lg').simulate('click');
  });
});

describe('<RegisterPage />', () => {
  it('should render component when email already exists', () => {
    const registerPage = mount(<BrowserRouter><RegisterPage /></BrowserRouter>);
    expect(registerPage.find('.reg-btn.box-sh-dark.hover-lg')).toHaveLength(1);
    expect(registerPage).toMatchSnapshot();
    registerPage.find('input[name="fullName"]').simulate('change', { target: {name: "fullName", value: "John" }});
    registerPage.find('input[name="lastname"]').simulate('change', { target: {name: "lastname", value: "Wick" }});
    registerPage.find('input[name="email"]').simulate('change', { target: {name: "email", value: "andela@gmail.com" }});
    registerPage.find('input[name="phone"]').simulate('change', { target: {name: "phone", value: "08131976306" }});
    registerPage.find('input[name="password"]').simulate('change', { target: {name: "password", value: "Ozulumba1234" }});
    registerPage.find('.reg-btn.box-sh-dark.hover-lg').simulate('click');
  });
});

describe('<RegisterPage />', () => {
  it('should render component when validation passes', () => {
    const registerPage = mount(<BrowserRouter><RegisterPage /></BrowserRouter>);
    expect(registerPage.find('input[name="fullName"]')).toHaveLength(1);
    expect(registerPage.find('input[name="lastname"]')).toHaveLength(1);
    expect(registerPage.find('input[name="email"]')).toHaveLength(1);
    expect(registerPage.find('input[name="phone"]')).toHaveLength(1);
    expect(registerPage.find('input[name="password"]')).toHaveLength(1);
    expect(registerPage.find('.reg-btn.box-sh-dark.hover-lg')).toHaveLength(1);
    expect(registerPage).toMatchSnapshot();
    registerPage.find('input[name="fullName"]').simulate('change', { target: {name: "fullName", value: "John" }});
    registerPage.find('input[name="lastname"]').simulate('change', { target: {name: "lastname", value: "Wick" }});
    registerPage.find('input[name="email"]').simulate('change', { target: {name: "email", value: randomEmail({ domain: 'gmail.com' }) }});
    registerPage.find('input[name="phone"]').simulate('change', { target: {name: "phone", value: "08131976306" }});
    registerPage.find('input[name="password"]').simulate('change', { target: {name: "password", value: "Ozulumba1234" }});
    registerPage.find('.reg-btn.box-sh-dark.hover-lg').simulate('click');
  });
});

