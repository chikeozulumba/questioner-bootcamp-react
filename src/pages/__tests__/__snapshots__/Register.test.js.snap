// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> should render component when email already exists 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RegisterPage>
      <withRouter(Home)>
        <Route>
          <Home
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="container"
            >
              <div
                className="card mauto m4 p3"
                id="login-form"
              >
                <form
                  action="/"
                  method="POST"
                >
                  <div
                    className="container"
                  >
                    <h3
                      className="md-text"
                    >
                      Register
                    </h3>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Full Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="fullName"
                        onChange={[Function]}
                        placeholder="Enter your Full Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Last Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="lastname"
                        onChange={[Function]}
                        placeholder="Last Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Email Address
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="email"
                        onChange={[Function]}
                        placeholder="Your Email Address"
                        type="email"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Phone Number
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="phone"
                        onChange={[Function]}
                        placeholder="Enter Phone Number"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Password
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="password"
                        onChange={[Function]}
                        placeholder="*************"
                        type="password"
                      />
                    </div>
                    <div
                      className="m2 input"
                    >
                      <a
                        className="reg-btn box-sh-dark hover-lg"
                        onClick={[Function]}
                      >
                        Join Questioner
                      </a>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </Home>
        </Route>
      </withRouter(Home)>
      <Footer>
        <footer
          className="bg-secondary"
        >
          <div
            className="container"
          >
            <div
              className="info-about"
            >
              <h3
                className="md-text"
              >
                Interested in Questioner?
              </h3>
              <div
                className="social-links"
              >
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/facebook.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/instagram.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/twitter.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/linkedin.svg"
                  />
                </a>
              </div>
            </div>
            <div
              className="additional-info"
            >
              <h3
                className="md-text"
              >
                Questioner Goals
              </h3>
              <p
                className="sm-text"
              >
                We create a safe space for others to ask for help. We ask for help when weâ€™re stuck and motivate others to do the same.
              </p>
              <p
                className="sm-text"
              >
                We create space for psychological safety in all our interactions by encouraging peers to be open and honest about their experiences.
              </p>
            </div>
          </div>
        </footer>
      </Footer>
    </RegisterPage>
  </Router>
</BrowserRouter>
`;

exports[`<RegisterPage /> should render component when validation fails 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RegisterPage>
      <withRouter(Home)>
        <Route>
          <Home
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="container"
            >
              <div
                className="card mauto m4 p3"
                id="login-form"
              >
                <form
                  action="/"
                  method="POST"
                >
                  <div
                    className="container"
                  >
                    <h3
                      className="md-text"
                    >
                      Register
                    </h3>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Full Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="fullName"
                        onChange={[Function]}
                        placeholder="Enter your Full Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Last Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="lastname"
                        onChange={[Function]}
                        placeholder="Last Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Email Address
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="email"
                        onChange={[Function]}
                        placeholder="Your Email Address"
                        type="email"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Phone Number
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="phone"
                        onChange={[Function]}
                        placeholder="Enter Phone Number"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Password
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="password"
                        onChange={[Function]}
                        placeholder="*************"
                        type="password"
                      />
                    </div>
                    <div
                      className="m2 input"
                    >
                      <a
                        className="reg-btn box-sh-dark hover-lg"
                        onClick={[Function]}
                      >
                        Join Questioner
                      </a>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </Home>
        </Route>
      </withRouter(Home)>
      <Footer>
        <footer
          className="bg-secondary"
        >
          <div
            className="container"
          >
            <div
              className="info-about"
            >
              <h3
                className="md-text"
              >
                Interested in Questioner?
              </h3>
              <div
                className="social-links"
              >
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/facebook.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/instagram.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/twitter.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/linkedin.svg"
                  />
                </a>
              </div>
            </div>
            <div
              className="additional-info"
            >
              <h3
                className="md-text"
              >
                Questioner Goals
              </h3>
              <p
                className="sm-text"
              >
                We create a safe space for others to ask for help. We ask for help when weâ€™re stuck and motivate others to do the same.
              </p>
              <p
                className="sm-text"
              >
                We create space for psychological safety in all our interactions by encouraging peers to be open and honest about their experiences.
              </p>
            </div>
          </div>
        </footer>
      </Footer>
    </RegisterPage>
  </Router>
</BrowserRouter>
`;

exports[`<RegisterPage /> should render component when validation passes 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RegisterPage>
      <withRouter(Home)>
        <Route>
          <Home
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="container"
            >
              <div
                className="card mauto m4 p3"
                id="login-form"
              >
                <form
                  action="/"
                  method="POST"
                >
                  <div
                    className="container"
                  >
                    <h3
                      className="md-text"
                    >
                      Register
                    </h3>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Full Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="fullName"
                        onChange={[Function]}
                        placeholder="Enter your Full Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Last Name
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="lastname"
                        onChange={[Function]}
                        placeholder="Last Name"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Email Address
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="email"
                        onChange={[Function]}
                        placeholder="Your Email Address"
                        type="email"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Phone Number
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="phone"
                        onChange={[Function]}
                        placeholder="Enter Phone Number"
                        type="text"
                      />
                    </div>
                    <div
                      className="m1 input"
                    >
                      <label
                        className="m1"
                        htmlFor="name"
                      >
                        Password
                      </label>
                      <input
                        className="p1 b-rsm"
                        name="password"
                        onChange={[Function]}
                        placeholder="*************"
                        type="password"
                      />
                    </div>
                    <div
                      className="m2 input"
                    >
                      <a
                        className="reg-btn box-sh-dark hover-lg"
                        onClick={[Function]}
                      >
                        Join Questioner
                      </a>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </Home>
        </Route>
      </withRouter(Home)>
      <Footer>
        <footer
          className="bg-secondary"
        >
          <div
            className="container"
          >
            <div
              className="info-about"
            >
              <h3
                className="md-text"
              >
                Interested in Questioner?
              </h3>
              <div
                className="social-links"
              >
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/facebook.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/instagram.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/twitter.svg"
                  />
                </a>
                <a
                  href="#!"
                >
                  <img
                    alt=""
                    src="./assets/imgs/icons/linkedin.svg"
                  />
                </a>
              </div>
            </div>
            <div
              className="additional-info"
            >
              <h3
                className="md-text"
              >
                Questioner Goals
              </h3>
              <p
                className="sm-text"
              >
                We create a safe space for others to ask for help. We ask for help when weâ€™re stuck and motivate others to do the same.
              </p>
              <p
                className="sm-text"
              >
                We create space for psychological safety in all our interactions by encouraging peers to be open and honest about their experiences.
              </p>
            </div>
          </div>
        </footer>
      </Footer>
    </RegisterPage>
  </Router>
</BrowserRouter>
`;
